psi4/src/psi4/dct/dct_compute.cc

```diff
@@ -93,22 +69,24 @@ double DCTSolver::compute_energy() {
         total_energy = compute_energy_UHF();
     }
 
+    // If we have a variational method, the density is free.
+    if (options_.get_bool("OPDM") == true || (orbital_optimized_ && options_.get_str("THREE_PARTICLE") == "NONE")) {
```

Comment:
Just out of interest, would computing the OPDM mess up the 3 particle energy correction?  Probably a stupid question - I'm not familiar with that correction.  Or is this the case that it would be something akin to a CCSD(T) energy that's computed, but the CCSD density is really what's provided?

Reply:
"it would be something akin to a CCSD(T) energy that's computed, but the CCSD density is really what's provided" is exactly right. If you remember the DCFT-06 gradient paper you were third author on, it's exactly the same scenario. The orbitals and amplitudes minimize an energy functional that isn't the one you care about, so computing the OPDM requires nasty response terms.